libraries/scripting/jvm/src/kotlin/script/experimental/jvm/impl/KJvmCompiledScript.kt

```diff
@@ -139,17 +140,17 @@ private fun CompiledScript.makeClassLoaderFromDependencies(baseClassLoader: Clas
     val processedClasspathElements = mutableSetOf<URL>()
     fun recursiveClassPath(res: Sequence<URL>, classLoader: ClassLoader?): Sequence<URL> =
         when (classLoader) {
-            null -> res
+            null, baseClassLoader -> res
```

Comment:
I would still process the `null` case, e.g. generating an error if we've got null.

Reply:
But it should be after `baseClassLoader` check, since it could be `null`, and it is a valid situation.

Reply:
@ligee I'm not sure that fallbackClassloader of DualClassLoader should necessarily have base classloader as a parent. Is it really so?

Reply:
As I can see from usages, it is always a thread context classloader that is a base classloader at the same time, so there is no appropriate test

Reply:
I'd say it should convert to the baseClassLoader, but since it is possible to replace it, it is safer to leave the code as you wrote it.
About the tests - there are some tests with baseClassLoader set to null, and that's probably it.